utils/web_scraper.py

```
@@ -288,51 +439,230 @@ async def _fetch_weibo_trending_fallback(limit: int = 10) -> Dict[str, Any]:
-            logger.error(f"B站爬取异常: {bilibili_result}")
-            bilibili_result = {'success': False, 'error': str(bilibili_result)}
+        headers = {
+            'User-Agent': get_random_user_agent(),
+            'Accept': 'application/json',
+            'Accept-Language': 'en-US,en;q=0.9',
+        }
         
-        if isinstance(weibo_result, Exception):
-            logger.error(f"微博爬取异常: {weibo_result}")
-            weibo_result = {'success': False, 'error': str(weibo_result)}
+        await asyncio.sleep(random.uniform(0.1, 0.3))
```

Ruff (0.14.10)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)